cli/src/cmd/run.rs

```diff
@@ -58,16 +57,86 @@ impl Cmd for RunArgs {
         #[cfg(not(feature = "sputnik-evm"))]
         unimplemented!("`run` does not work with EVMs other than Sputnik yet");
 
+        let mut evm_opts = self.evm_opts.clone();
+        if evm_opts.debug {
+            evm_opts.verbosity = 3;
+        }
+
+        // This is just the contracts compiled, but we need to merge this with the read cached
+        // artifacts
+
         let func = IntoFunction::into(self.sig.as_deref().unwrap_or("run()"));
-        let (abi, bytecode, _) = self.build()?.into_parts();
+        let BuildOutput { contract, mut highlevel_known_contracts, sources } = self.build()?;
+
+        // if we have a high verbosity, we want all possible compiler data not just for this
+        // contract in case the transaction interacts with others
+        if evm_opts.debug || evm_opts.verbosity > 3 {
+            let project = self.opts.project()?;
+            println!("compiling broader repo...");
+            let output = project.compile()?;
+            if output.has_compiler_errors() {
+                // return the diagnostics error back to the user.
+                eyre::bail!(output.to_string())
+            } else if output.is_unchanged() {
+                println!("no files changed, compilation skippped.");
+            } else {
+                println!("success.");
+            }
```

Comment:
I just noticed, but adding this introduces the requirement of having a full directory structure.

Cloning and running the following will fail:

 
```
 RUST_LOG=forge=trace,evm-adapters=trace cargo r --bin forge -- run ./run_test.sol --rpc-url $ETH_RPC_URL --debug
compiling...
success.
Error:
   0: "/Users/gakonst/oss/foundry/lib": No such file or directory (os error 2)
```

Then, when I tried using it on a repo generated by `forge init`, it didn't seem to resolve remappings (which seems weird, given that we're compiling using the solc toolchain.


```
forge init foo
cd foo
forge t # this works
forge run ./src/test/Contract.t.sol --sig "testExample()" # this won't work because it doesn't use the full compilation toolchain
forge run ./src/test/Contract.t.sol --sig "testExample()" --debug # this fails with the same error, when i'd expect that it'd work because it uses ethers-solc's compile
```

Reply:
oops BuildArgs should be optional i believe and we should handle that - good catch

Reply:
this could be an issue in ethers-solc as well, reading files from non existing lib directories should not fail

Reply:
i just needed better error handling on `self.opts.project` instead of using the `?`. fixed in local branch